core/src/main/java/org/infinispan/stats/impl/ClusterContainerStatsImpl.java

```diff
@@ -55,25 +46,9 @@ public void start() {
    }
 
    @Override
-   void updateStats() throws Exception {
-      List<Map<String, Number>> memoryMap = getClusterStatMaps();
-      for (String att : LONG_ATTRIBUTES)
-         putLongAttributes(memoryMap, att);
-   }
-
-   private List<Map<String, Number>> getClusterStatMaps() throws Exception {
+   protected List<Map<String, Number>> statistics() throws Exception {
       final List<Map<String, Number>> successfulResponseMaps = new ArrayList<>();
-      CompletableFutures.await(clusterExecutor.submit(() -> {
-         Map<String, Number> map = new HashMap<>();
-         long available = Runtime.getRuntime().freeMemory();
-         long total = Runtime.getRuntime().totalMemory();
-         long max = Runtime.getRuntime().maxMemory();
-         map.put(MEMORY_AVAILABLE, available);
-         map.put(MEMORY_MAX, max);
-         map.put(MEMORY_TOTAL, total);
-         map.put(MEMORY_USED, total - available);
-         successfulResponseMaps.add(map);
-      }));
+      CompletableFutures.await(clusterExecutor.submit(() -> successfulResponseMaps.add(getLocalStatMaps())));
```

Comment:
This will not work in different JVM as the ArrayList is serialized to the remote node but no way to get the result back.
You can use the `submitConsumer` method which allows you to return a Map from the function which is serialized and returned back and passed to the TriConsumer which can collect the results.